src/components/Button/Button.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { FC, useMemo } from 'react';
+
+import { TouchableOpacityProps, ActivityIndicator } from 'react-native';
+import { variant, space, layout } from 'styled-system';
+import styled from 'styled-components/native';
+
+import { theme } from 'src/theme';
+import { Text, ColumnProps, Icon, Row } from 'src/components';
+
+const PRIMARY = 'primary';
+const SECONDARY = 'secondary';
+const DISABLED = 'disabled';
+
+interface ButtonComponentProps extends ColumnProps, TouchableOpacityProps {
+  title: string;
+  icon?: string;
+  variant: string;
+  disabled?: boolean;
+  isLoading?: boolean;
+  accessibilityLabel: string;
+}
+interface StyledButtonProps extends TouchableOpacityProps {
+  variant: string;
+}
+const ButtonComponent: FC<ButtonComponentProps> = ({
+  title,
+  variant,
+  disabled,
+  isLoading,
+  icon,
+  ...props
+}) => {
+  const { colors } = theme;
+
+  const textColor = useMemo(
+    () => (variant === PRIMARY ? 'white' : variant === SECONDARY ? 'black' : 'gray.n500'),
+    [variant, disabled, colors]
+  );
+
+  return (
+    <StyledButton variant={disabled ? 'disabled' : variant} disabled={disabled} {...props}>
+      {isLoading && <ActivityIndicator size='small' color='white' />}
+
+      {!isLoading && (
+        <Row>
+          {icon && <Icon icon={icon} marginRight={13} color={textColor} />}
+
+          <Text variant='small' color={textColor} fontWeight={600} lineHeight='24px'>
+            {title}
+          </Text>
+        </Row>
+      )}
+    </StyledButton>
+  );
+};
+
+const StyledButton: FC<StyledButtonProps> = styled.TouchableOpacity(
+  variant({
+    variants: {
+      [PRIMARY]: {
+        backgroundColor: 'primary',
+        borderColor: 'transparent'
+      },
+      [SECONDARY]: {
+        backgroundColor: 'transparent',
+        borderColor: 'black'
+      },
+      [DISABLED]: {
+        backgroundColor: 'gray',
```

Comment:
Ao invés de modificar as cores, para o caso de disabled não faria mais sentido apenas diminuir talvez o opacity do botão? Pois, se for um botão do tipo **PRIMARY** ao desabilitá-lo em algum momento, ele vai modificar a cor, causando um desconforto no usuário.

Reply:
Não, se for pra desabilitar o botão, além de setar o disabled como true. Setar um opacity de 0.7 por exemplo no botão. Dá uma procurada sobre opacity css

Reply:
Aliás, acho que falei asneira aqui, mano hehe
Perdão!
Mas acho que você poderia colocar talvez uma cor apenas um pouco mais clara. Ou até mesmo ver se o opacity funciona.
E o que eu tô falando é a prop opacity de css, caso surja alguma dúvida, me dá um grito que a gente conversa

Reply:
Então, os dois funcionam relativamente bem. Acho que se eu baixar um tom de cinza o botão fica numa cor boa, olha:

![image](https://user-images.githubusercontent.com/37683337/121700400-3f480b00-caa6-11eb-8539-f3058c602191.png)

mesmo com opacity de '0.5'  ainda fica escuro no botao primary, por ser a cor preta. 

![image](https://user-images.githubusercontent.com/37683337/121700584-6ef71300-caa6-11eb-8fc0-8d1d416afa97.png)

Acho que mudar um tom para cinza mais claro fique um pouco melhor nesse caso.